src/http/web_socket.cr

```diff
@@ -100,52 +101,109 @@ class HTTP::WebSocket
     @ws.close(message)
   end
 
-  def run
+  enum Opcode
```

Comment:
Why not reuse `HTTP::WebSocket::Protocol::Opcode` for this? `Protocol` is internal API, but you could simply move that enum here and use it in both classes. The `CONTINUATION` value is not needed for this high-level API, but I don't think having that is a huge issue.
But I don't know what others think about this.

Reply:
Actually, I was thinking about this too.

Mostly I was concerned about the `CONTINUATION` and the fact that the enum names are all-caps instead of title-cased. It makes it seem like it's for a low-level API (IMO). I'm not sure what the convention is within Crystal.

But I also like to reuse things if possible. Personally I think I lean more towards reusing the enum but wanted feedback from people more familiar with the language. Thanks for your input.

Reply:
> enum names are all-caps instead of title-cased.

There ist actually no style guide preferring either of them. And this could easily be adapted because the current enum is just internal API.

Reply:
I moved the enum into WebSocket class and imported it from the Protocol. I left it as all-caps in lieu of a style guide.